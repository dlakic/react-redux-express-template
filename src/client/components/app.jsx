import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { hot } from 'react-hot-loader';
import { changeColor } from '../store/action';
import ReduxDevTools from './redux-dev-tools';

export const App = ({ color, dispatch }, { store }) => {
  function handleClick(chosenColor) {
    dispatch(changeColor(chosenColor));
  }

  let DevTools = null;
  if (process.env.NODE_ENV === 'development') {
    DevTools = <ReduxDevTools store={store} />;
  }

  return (
    <div>
      <h1>Welcome to the react-redux test app!</h1>
      <p className={color}>It works!</p>
      <button type="button" onClick={() => handleClick('green')}>green</button>
      <button type="button" onClick={() => handleClick('yellow')}>yellow</button>
      <button type="button" onClick={() => handleClick('red')}>red</button>
      {DevTools}
    </div>
  );
};

App.contextTypes = {
  store: PropTypes.shape({
    color: PropTypes.string.isRequired,
  }),
};

App.propTypes = {
  dispatch: PropTypes.func.isRequired,
  color: PropTypes.string.isRequired,
};

function mapStateToProps(state) {
  return {
    color: state.color,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(App));
