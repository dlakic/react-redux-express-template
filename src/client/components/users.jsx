import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchUsers } from '../store/action';

export class Users extends Component {
  constructor(props) {
    super(props);

    this.state = {
      newUserName: '',
      newUserProfession: '',
    };

    this.props.fetchUsers();
  }

  changeName(name) {
    this.setState({
      newUserName: name,
    });
  }

  changeProfession(profession) {
    console.log(profession);
    this.setState({
      newUserProfession: profession,
    });
  }

  addUser(e) {
    e.preventDefault();
  }

  render() {
    const { users, userLoading } = this.props;

    if (userLoading) {
      return (<div className="users-container"><h2> Loading ... </h2></div>);
    }

    return (
      <div className="users-container">
        <h2>Users on Mock DB:</h2>
        {users.map(user => (
          <div className="user-container">
            <p>
              Name:&nbsp;
              {user.name}
            </p>
            <p>
              Profession:&nbsp;
              {user.profession}
            </p>
          </div>
        ))}
        <h2> Add new user to Mock DB </h2>
        <label className="user-input-label" htmlFor="username">
          Username:
          <input type="text" name="username" onChange={e => this.changeName(e.target.value)} />
        </label>
        <label className="user-input-label" htmlFor="profession">
          Profession:
          <input type="text" name="profession" onChange={e => this.changeProfession(e.target.value)} />
        </label>
        <button onClick={e => this.addUser(e)}>Add new User</button>
      </div>
    );
  }
}

Users.propTypes = {
  users: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  fetchUsers: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
  return {
    users: state.users,
    userLoading: state.userLoading,
  };
}

const mapDispatchToProps = { fetchUsers };
export default connect(mapStateToProps, mapDispatchToProps)(Users);
