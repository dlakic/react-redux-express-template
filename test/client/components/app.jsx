import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from '../../testutils';
import { App } from '../../../src/client/components/app';

test('Rendering the entire app', () => {
  const props = {
    color: 'green',
    dispatch: () => {},
  };

  const component = renderer.create(
    <Provider>
      <App {...props} />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
