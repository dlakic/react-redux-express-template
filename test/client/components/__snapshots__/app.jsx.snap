// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the entire app 1`] = `
<div>
  <h1>
    Welcome to the react-redux test app!
  </h1>
  <p
    className="green"
  >
    It works!
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    green
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    yellow
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    red
  </button>
</div>
`;
